text/2018-11-12-unified-key-format.md

```diff
@@ -0,0 +1,93 @@
+# Summary
+
+This RFC proposes a new unified and human readable format for keys. All TiDB software stack
+components (e.g. TiDB, TiKV, pd-ctl, tikv-ctl, ...) *must* follow this new format when printing out
+keys and accepting keys from user input like command line parameters.
+
+# Motivation
+
+Currently we have multiple human readable formats for keys:
+
+- Hex format, like `74000000000000001C5F7200000000000000FA`
+
+- Protobuffer format, like `t\000\000\000\000\000\000\000\034_r\000\000\000\000\000\000\000\372`
+
+- Go format, like `t\x00\x00\x00\x00\x00\x00\x00\x1c_r\x00\x00\x00\x00\x00\x00\x00\xfa`
+
+Defects:
+
+- Bad interoperability: These different formats are pain points when working with different tools
+  together. For example, pd-ctl produces keys in Go format while tikv-ctl accepts keys in
+  Protobuffer format.
+
+- Hard to implement: Actually there are some attempts to support different kind of formats, like
+  https://github.com/tikv/tikv/pull/3182 and https://github.com/pingcap/pd/pull/474. However it
+  turns out that these implementations won't work in real world scenario. For example, it is
+  not sufficient to decode the Golang format by only dealing with hex escapes `\x..` [\[1\]][1]
```

Comment:
I feel like the content within the code block is not what you intended here? Same with line below.

Reply:
Hmm.. It is actually. For example, I wanted to say that, the go format is not simply "\xXX" (but also contains other complicated escape sequence like "\uXX" and "\UXXXX").

Reply:
I actually realized later this was a reference link and forgot to delete it. Thanks for pointing it out. :)